pkg/geo/geomfn/add_measure.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+func AddMeasureToLineString(
+	geometry geo.Geometry, start float64, end float64,
+) (geo.Geometry, error) {
+	t, err := geometry.AsGeomT()
+	if err != nil {
+		return geometry, err
+	}
+
+	switch t := t.(type) {
+	case *geom.LineString: //, *geom.MultiLineString: TODO have a function that calls this multiple times
+		// Add an M dimension to the linestring if it doesn't have one already.
+		geometryWithM, err := ForceLayout(geometry, augmentLayoutWithM(t.Layout()))
+		if err != nil {
+			return geometry, nil
+		}
+		newT, err := geometryWithM.AsGeomT()
+		if err != nil {
+			return geometry, err
+		}
+		newLineString, ok := newT.(*geom.LineString)
+		if !ok {
+			return geometry, errors.Newf("ForceLayout unexpectedly transformed a LineString to a different geometry")
+		}
+
+		// Handle special case where all the measure values will be the same.
+		if start == end {
+			newT, err := applyOnCoordsForGeomT(newLineString, func(l geom.Layout, dst []float64, src []float64) error {
+				copy(dst, src)
+				dst[newLineString.Layout().MIndex()] = start
+				return nil
+			})
+			if err != nil {
+				return geometry, err
+			}
+			return geo.MakeGeometryFromGeomT(newT)
+		}
+
+		// Compute 2D length of linestring.
+		lineLength, err := Length(geometry)
```

Comment:
the call to `Length` is also expensive - you need to reserialize an EWKB, go cgo to GEOS and come back. the calculation is relatively simple for distance between two points. we may even have it in `geodist`. can you use that instead?

Reply:
i think `coordNorm(coordSub(b, a))` should do it.

Reply:
> i think `coordNorm(coordSub(b, a))` should do it.

I think this would work for replacing the calculation of the distance between two points but we also still need to calculate the actual length of the line (not just the distance between the start and end point).

Reply:
Well actually I guess I can just sum up the distance between each pair of points first

Reply:
Oh right. Yeah it's ok to use length for the overall calculation, my bad on reading that. That said you can do as you suggested!